Automattic/woocommerce-subscriptions-core
# Add support for unit tests in php8+

Fixes #139 

## Description

Adds support for unit testing in php8.0 and 8.1. This also removes the ability to run unit tests on php7.0
(compatibility with php7.0 still exists).

PHPUnit has been updated to the latest version which allows us to test against php7.1 - php8.1. Github actions have also been updated to run the test suite against all supported versions.

As part of this PR I encountered issues with the current github actions were testing against the oldest version for WC which was causing issues on newer versions of php. 
So github actions have been updated so we now run unit tests on the latest version of WP and the following versions of WC / php

WC Version | PHP Version | Notes
---|---|---
Latest | 7.1 - 8.1 |
Beta | 7.4 | Any version would do. I went with middle of the road for compatibility broad compatibility
5.5 | 7.1 | This matches the oldest version of WC we currently test


## How to test this PR

Setup unit tests as mentioned in `tests/README.md` and confirm they all pass. Additionally github actions output can be viewed.

- [x] Will this PR affect WooCommerce Subscriptions? no, developer only change.
- [x] Will this PR affect WooCommerce Payments? no, developer only change.
